ui/src/app/views/workspace/component.ts

```diff
@@ -292,33 +292,33 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
     this.router.navigate(['share'], {relativeTo : this.route});
   }
 
-  sendANotebook(notebook): void {
-    let enableBtn = false;
-    let unSelectCheckBox = false;
+  selectANotebook(notebook): void {
+    let oneTrueExist = false;
```

Comment:
Could you use the code I provided in my other comment or else provide a response to my comment on this? This function is still way too verbose in my opinion. I would not worry about making 2 passes instead of 1 at this point; if we have 1000s of notebooks we have bigger UX issues here.